search/api.py

```diff
@@ -133,6 +141,55 @@ def gen_video_id(video_obj):
     return "video_{}_{}".format(video_obj.platform, video_obj.video_id)
 
 
+def gen_doc_ids(items):
+    """
+    Return a list of ES document ids for user favorites or list items
+
+    Args:
+        items(list of FavoriteItem or UserListItem): a user's favorites/list items
+
+    Returns:
+        list of str: the ES document ids for the favorited/list items
+
+    """
+    doc_ids = []
+    for item in filter(lambda i: i.item is not None, items):
+        classname = item.content_type.name
+        if classname == COURSE_TYPE:
+            doc_ids.append(gen_course_id(item.item.platform, item.item.course_id))
+        elif classname == BOOTCAMP_TYPE:
+            doc_ids.append(gen_bootcamp_id(item.item.course_id))
+        elif classname == PROGRAM_TYPE:
+            doc_ids.append(gen_program_id(item.item))
+        elif classname == VIDEO_TYPE:
+            doc_ids.append(gen_video_id(item.item))
+        elif classname == USER_LIST_TYPE:
+            doc_ids.append(gen_user_list_id(item.item))
+    return doc_ids
+
+
+def gen_lists_dict(user):
+    """
+    Return a dict of list ids for each item in a user's lists
+
+    Args:
+        user(User): a user to retrieve UserListItems for
+
+    Returns:
+        dict: a mapping of list ids (value) per resource item ES id (key)
+
+    """
+    lists_dict = collections.defaultdict(list)
+    items = (
+        UserListItem.objects.filter(user_list__author=user)
```

Comment:
Suggest we add a `.only("id", "user_list_id")` here so we're only fetching the fields we absolutely need to return the result.